sense-android-library/src/nl/sense_os/datastorageengine/CommonSenseProxy.java

```diff
@@ -0,0 +1,597 @@
+            return -2;
+        } else if (!"200".equalsIgnoreCase(responseCode)) {
+            Log.w(TAG, "CommonSense" + method + "failed! Response: " + responseCode);
+            return -1;
+        } else {
+            // received 200 response
+            Log.e(TAG, "CommonSense" + method + "OK! ");
+            return  0;
+        }
+    }
+
+
+    /**
+     * Make a url with the included action
+     *
+     * @param action	The required endpoint
+     * @param appendix	The appendix of the endpoint, can be null
+     *
+     * @result	The required url
+     */
+    private static String makeCSRestUrlFor(String action, String appendix)
+    {
+        if(action == null || action.isEmpty()){
+            return null;
+        }
+        if(appendix == null)
+            appendix = "";
+        String url = URL_BASE + "/" + action + URL_JSON_SUFFIX + appendix;
+        return url;
+    }
+
+    /**
+     * Make a url with the included action
+     *
+     * @param action	The required endpoint
+     * @param appendix	The appendix of the endpoint, can be null
+     *
+     * @result	The required url
+     */
+    private static String makeUrlFor(String action, String appendix)
+    {
+        if(action == null || action.isEmpty()){
+            return null;
+        }
+        if(appendix == null)
+            appendix = "";
+        String url;
+        if(action.equals(URL_LOGIN) || action.equals(URL_LOGOUT)){
+            url = URL_AUTH + "/" + action + appendix;
+        }else{
+            url = URL_BASE + "/" + action + URL_JSON_SUFFIX + appendix;
+        }
+
+        return url;
+    }
+
+    /**
+     * Helper function for getting a list from the cs-rest API
+     *
+     * @param urlAction     Part og url.
+     * @param paramsString  Part of url.
+     * @param resultKey     The name of the query field.
+     * @param sessionID     The sessionID of the current user.
+     * @param pageSize      The number of items in one page.
+     * @param methodName    The name of the caller method.
+     *                      
+     * @throws IOException, JSONException
+     * @result	The integer value of the result
+     *
+     */
+   private static JSONArray getListForURLAction(String urlAction, String paramsString, String resultKey, String sessionID, int pageSize, String methodName) throws IOException, JSONException
+   {
+       int page = 0;
+       JSONArray resultList = new JSONArray();
+       boolean done = false;
+       while (!done) {
+           String params = "?page=" + page + paramsString;
+           String url =  makeUrlFor(urlAction,params);
+           Map<String, String> response = request(url, null, sessionID, HTTP_METHOD_GET);
+
+           int codeResult = checkResponseCode(response.get(RESPONSE_CODE), methodName);
+           if (codeResult != 0) {
+               throw new IOException("Incorrect response of " + methodName + " from CommonSense,\n the content is:" + response.get(RESPONSE_CONTENT));
```

Comment:
@excaliburnan throw an appropriate response, so it can be handled the right way. Maybe add the response code. It's good to know what the response code is because when the session id is expired we want to know this because a relogin is then required.